execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -894,23 +896,23 @@ where
         let genesis_delegators: Vec<_> = self.exec_config.get_bonded_delegators().collect();
 
         // Make sure all delegators have corresponding genesis validator entries
-        for (&validator_public_key, &delegator_public_key, _balance, delegated_amount) in
-            &genesis_delegators
+        for (validator_public_key, delegator_public_key, _balance, delegated_amount) in
+            genesis_delegators.clone().into_iter()
```

Comment:
```suggestion
            genesis_delegators.iter()
```

You are cloning below anyway, no need to clone here.

Reply:
Right, but then it would be moved here then. I should ideally remove the following clone then

Reply:
@xcthulhu I got it working with `iter`